test/unit/org/apache/cassandra/db/commitlog/BatchCommitLogTest.java

```diff
@@ -75,4 +81,22 @@ public void testBatchCLShutDownImmediately() throws InterruptedException
         Assert.assertTrue("Expect batch commitlog shutdown immediately, but took " + delta, delta < CL_BATCH_SYNC_WINDOW);
         CommitLog.instance.start();
     }
+
+    @Test
+    public void failWrite() throws IOException
+    {
+        ColumnFamilyStore cfs = Keyspace.open(KEYSPACE1).getColumnFamilyStore(STANDARD1);
```

Comment:
this test has two flaws:
1. doesn't test that writes are not failed until a failed sync. FWIW an always-throwing `add` implementation happily passes the test
2. more importantly - we're testing a scenario "in specific internal state of the commit log the new writes are failed", whereas we should be testing "after a failed sync writes should be failed"

This 2nd point sounds hard and expensive to implement, but I think it is not necessarily the case. Please check out `@RunWith(BMUnitRunner.class)`, `@BMRules` etc. annotations as in e.g. `CompactionsBytemanTest`. I see reasonable potential to make a simple test in which `commitLog.sync` throws, and we see failing writes until a subsequent, successful sync.